src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -91,20 +92,46 @@ public void setupSchema(List<ColumnMetadata> columns) {
       if (!column.getNullable()) {
```

Comment:
1. the ParquetColumn.columnIndex now doesn't map to either the fieldId or the columnOrdinal, is that correct?
2. How do the IDs flow if I have a table with schema (int, struct, string, array, float, map) ? Are we assuming that all subcolumns are meant to be at the end of the list always?

Reply:
1. Yes. The `ParquetColumn.columnIndex` is used for internal check and starts from zero. While `columnOrdinal` and `fieldId` starts from one.

Reply:
lets chat about this f2f. I see the ParquetColumn.index / this.fieldIndex are used in addRow and it looks like they're meant to only track "top level" fields in terms of what data is sent in by the customer, wherease with your changes they will end up being much larger than necessary.
For instance lets take a table with a single structured object column (C1) with 100 fields (F1-F100), and each field is a structured object with 100 fields each (FF1-FF100).
In addRow you'll get a Map with a single entry (key=C1, value = Map-with-100-entries).
However you'll end up allocating an array with ~100*100 = 10k elements because of this field id business.

Reply:
pl checkout the comment here too, on this.fieldIndex.

Reply:
`this.fieldIndex` is used for row validation with top level column, and the `fieldIndex` only contains top level columns.